lib/cocoapods/command/repo/add.rb

```diff
@@ -39,16 +39,48 @@ def validate!
         def run
           prefix = @shallow ? 'Creating shallow clone of' : 'Cloning'
           UI.section("#{prefix} spec repo `#{@name}` from `#{@url}`#{" (branch `#{@branch}`)" if @branch}") do
-            config.repos_dir.mkpath
-            Dir.chdir(config.repos_dir) do
-              command = ['clone', @url, @name]
-              command << '--depth=1' if @shallow
-              git!(command)
-            end
-            Dir.chdir(dir) { git!('checkout', @branch) } if @branch
+            create_repos_dir
+            clone_repo
+            checkout_branch
             SourcesManager.check_version_information(dir)
           end
         end
+
+        private
+
+        # Creates the repos directory specified in the configuration by
+        # `config.repos_dir`.
+        #
+        # @return [void]
+        #
+        # @raise  If the directory cannot be created due to a system error.
+        #
+        def create_repos_dir
+          config.repos_dir.mkpath
+        rescue SystemCallError => e
+          raise Informative, "Could not create repos directory '#{config.repos_dir}'.\n#{e.class.name}: #{e.message}"
```

Comment:
Could this say `"Could not create '#{config.repos_dir}', the CocoaPods repo cache directory.\n#{e.class.name}: #{e.message} ` - I think it's a bit clearer about why that folder has to exist.